docs/flashbots-data/mev-inspect-py/architecture.md

```diff
@@ -0,0 +1,36 @@
+---
+title: architecture
+---
+
+#### Classified Traces
+
+For each block we intend to inspect, we first fetch all of its traces, transaction receipts, and other additional information. The raw traces are then processed into classified traces (with protocol name, relevant function signature, relevant call inputs, strategy classification and other information) before being passed onto individual inspectors. 
+
+If we notice these classified traces to only contain liquidation related functions, we can only pass them off to aave/comp inspectors. Similarly, arbitrage profits are reduced by running them through protocol inspectors tagged in this stage (swap/liquidate/buy tx followed by addLiquidity etc). 
+
+#### Strategy Inspectors
+
+Each strategy has its own inspector and we define the types based on what output we expect it to return. This could include net profits, but also other information such as whether it was a pre-flight check (querying the reserves to see if the arb is still available) or a successful mev opportunity. 
+
+TODO: generic types we've narrowed them down to
+TODO: table of inspectors pending/wip/ready
+
+#### Tokenflow
```

Comment:
Tokenflow getting its own section in architecture is good. What about summarizers?